project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit, Inject, OnDestroy } from '@angular/core'
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog'
+import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { HttpErrorResponse } from '@angular/common/http'
+import { MatSnackBar } from '@angular/material'
+
+import { Subject } from 'rxjs'
+import { takeUntil } from 'rxjs/operators'
+
+import { UserProfileService } from '../../../user-profile/services/user-profile.service'
+import { ConfigurationsService } from '@sunbird-cb/utils/src/public-api'
+
+@Component({
+  selector: 'ws-transfer-request',
+  templateUrl: './transfer-request.component.html',
+  styleUrls: ['./transfer-request.component.scss'],
+})
+
+export class TransferRequestComponent implements OnInit, OnDestroy {
+
+  transferRequestForm = new FormGroup({
+    organization: new FormControl('', [Validators.required]),
+    group: new FormControl('', [Validators.required]),
+    designation: new FormControl('', [Validators.required]),
+  })
+  profileMetaData: any
+  departmentData: any[] = []
+  private destroySubject$ = new Subject()
+  groupData: any[] = []
+
+  constructor(
+    public dialogRef: MatDialogRef<TransferRequestComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private userProfileService: UserProfileService,
+    private matSnackBar: MatSnackBar,
+    private configService: ConfigurationsService
+  ) {
+    if (this.data.portalProfile.professionalDetails && this.data.portalProfile.professionalDetails.length) {
+      this.transferRequestForm.controls.group.setValue(this.data.portalProfile.professionalDetails[0].group)
+      this.transferRequestForm.controls.designation.setValue(this.data.portalProfile.professionalDetails[0].designation)
+    }
+    if (this.data.portalProfile.employmentDetails) {
+      this.transferRequestForm.controls.organization.setValue(this.data.portalProfile.employmentDetails.departmentName)
+    }
+  }
+
+  ngOnInit() {
+    this.getGroupData()
+    this.getAllDeptData()
+    this.getProfilePageMetaData()
+  }
+
+  handleCloseModal(): void {
+    this.dialogRef.close()
+  }
+
+  handleSubmitRequest(): void {
+    const data: any = {
+      'name': this.transferRequestForm.value['organization'],
+      'designation': this.transferRequestForm.value['designation'],
+      'group': this.transferRequestForm.value['group'],
+    }
+    const postData: any = {
+      'request': {
+        'userId': this.configService.unMappedUser.id,
+        'employmentDetails': {
+          'departmentName': this.transferRequestForm.value['organization'],
+        },
```

Comment:
please document, all these functions

Reply:
Let me close all the functionalities and then I will document each and every function